stl/inc/chrono

```diff
@@ -3521,6 +3526,1579 @@ namespace chrono {
+                    string _Name;
+                    _State |= _Get_Tz_Name(_First, _Ctype_fac, _Name);
+                    if (!_Update(_Tz_name, _Name)) {
+                        _State |= ios_base::failbit;
+                    }
+                    break;
+                }
+
+            default:
+                {
+                    // Invalid flag
+                    _State |= ios_base::failbit;
+                }
+            }
+
+            return _First;
+        }
+
+        template <class _InIt>
+        _InIt _Parse_time_field_restricted(_InIt _First, ios_base& _Iosbase, ios_base::iostate& _State,
+            const char* _Fmt, unsigned int _Subsecond_precision = 0) {
+            using _Ctype = ctype<typename _InIt::value_type>;
+            // Parses a restricted format string. It generally doesn't handle anything parsed outside of
+            // _Parse_time_field:
+            //   (a) any whitespace (' ', %n, %t)
+            //   (b) %% literal (other literals are parsed, however)
+            //   (c) E or O modifiers
+            //   (d) width parameter
+            // It also assumes a valid format string, specifically that '%' is always followed by a flag.
+            const _Ctype& _Ctype_fac{_STD use_facet<_Ctype>(_Iosbase.getloc())};
+            constexpr _InIt _Last{};
+
+            while (*_Fmt != '\0' && _First != _Last && _State == ios_base::goodbit) {
+                if (*_Fmt == '%') {
+                    _First = _Parse_time_field(_First, _Iosbase, _State, *++_Fmt, '\0', 0, _Subsecond_precision);
+                } else if (_Ctype_fac.narrow(*_First++) != *_Fmt) {
+                    _State |= ios_base::failbit;
+                }
+                ++_Fmt;
+            }
+            return _First;
+        }
+
+        template <class _InIt>
+        ios_base::iostate _Get_fixed(_InIt& _First, unsigned int _Width,
+            const ctype<typename _InIt::value_type>& _Ctype_fac,
+            const numpunct<typename _InIt::value_type>& _Numpunct_fac) {
+            constexpr _InIt _Last{};
+            int _Second_        = 0;
+            int64_t _Subsecond_ = 0;
+            int64_t _Multiplier = _SubsecondType::period::den;
+            bool _Found_point   = false;
+            bool _Found_digit   = false;
+
+            while (_First != _Last && _Width > 0 && _Multiplier >= 10) {
+                const auto _Ch = *_First++;
+                --_Width;
+                if (_Ch == _Numpunct_fac.decimal_point()) {
+                    _Found_point = true;
+                } else if (_Ctype_fac.is(ctype_base::digit, _Ch)) {
+                    _Found_digit      = true;
+                    const char _Digit = _Ctype_fac.narrow(_Ch);
+                    if (_Found_point) {
+                        _Multiplier /= 10;
+                        _Subsecond_ += (_Digit - '0') * _Multiplier;
+                    } else {
+                        _Second_ = _Second_ * 10 + (_Digit - '0');

```

Comment:
I can guard this for overflow, but it raises a related question: should we allow parsing more than 60 seconds (or 59 minutes or 23 hours) for the `duration` overload? I'm currently always limiting those fields to the indicated values, but maybe that only makes sense for the `time_point` overloads.

Reply:
It seems to me that there should be no limits for `duration` values - e.g. `%j` should be allowed to specify `days{1000}`, and similarly hours/minutes/seconds should be unlimited. Otherwise there should be Standardese specifying that there are limits (I find the `duration` behavior to be underspecified).